webapp/channels/src/entry.tsx

```diff
@@ -77,3 +77,26 @@ appendOnDOMContentLoadedEvent(() => {
     // Do the pre-render setup and call renderReactRootComponent when done
     preRenderSetup(renderReactRootComponent);
 });
+
+const registerServiceWorker = async () => {
+    if ('serviceWorker' in navigator) {
+        try {
+            const registration = await navigator.serviceWorker.register('/sw.js');
+            if (registration.installing) {
+                // eslint-disable-next-line no-console
+                console.log('Service worker installing');
+            } else if (registration.waiting) {
+                // eslint-disable-next-line no-console
+                console.log('Service worker installed');
+            } else if (registration.active) {
+                // eslint-disable-next-line no-console
+                console.log('Service worker active');
+            }
+        } catch (error) {
+            // eslint-disable-next-line no-console
+            console.error(`Service worker registration failed with ${error}`);
```

Comment:
If I understand right, this is going to error out for anyone using plain HTTP for their server. That'll likely break other things like notifications as well, but I wonder if we should check if the page is HTTP or HTTPS before trying to register the service worker to avoid adding an extra error to that

Reply:
Let me verify that.